core/src/data_sources/folder.rs

```diff
@@ -66,31 +53,3 @@ impl Folder {
         &self.parents
     }
 }
-
-impl From<Node> for Folder {
-    fn from(node: Node) -> Self {
-        Folder::new(
-            node.project(),
-            node.data_source_id(),
-            node.node_id(),
-            node.timestamp(),
-            node.title(),
-            node.parents().clone(),
-        )
-    }
-}
-
-impl From<Folder> for Node {
```

Comment:
How come we remove these?

Reply:
Explained in the OP, they are not needed. 
The only one we actually need is Node -> Folder, but we can't have it as an external trait implementation because the members are private, and the getters return refs (so we would need to clone each member of Node to get our Folder). 
I replaced this with a `into_folder()` method that consumes the `Node` into a `Folder`

Reply:
👍